Drone/drone_controller.py

```diff
@@ -103,7 +106,7 @@ def _update(self):
 
         if isinstance(self._task, HoverTask) or self._task is None:
             # Process remaining instructions
-            if len(self._instruction_queue):
```

Comment:
I would prefer if you would use the `len()` function actually
https://kushaldas.in/posts/len-function-in-python.html

Reply:
Your right, `len()` would be more readable than calling a size property. I just had an idea though. What are your thoughts on me adding an `empty()` method to the PriorityQueue object? Then the code could read `if not self._instruction_queue.empty()`, which has a really clear intent. (I will still overload `len()`, by the way)